pkg/policy/directory/watcher.go

```diff
@@ -34,12 +35,7 @@ type policyWatcher struct {
 	fileNameToCnpCache map[string]*cilium_v2.CiliumNetworkPolicy
 }
 
-func (p *policyWatcher) translateToCNPObject(file string) (*cilium_v2.CiliumNetworkPolicy, error) {
-	data, err := os.ReadFile(file)
-	if err != nil {
-		return nil, err
-	}
-
+func (p *policyWatcher) translateToCNPObject(data []byte) (*cilium_v2.CiliumNetworkPolicy, error) {
```

Comment:
Have you considered mocking the filesystem with https://pkg.go.dev/io/fs#FS

This way, you could pass in a filepath and a `fs.FS` reference here. You can obtain a real FS via https://pkg.go.dev/os#DirFS and for testing us a  https://pkg.go.dev/testing/fstest#MapFS ?

Reply:
will take a look